tests/Feature/DynamicButtonPropertyTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use PowerComponents\LivewirePowerGrid\Button;
+use PowerComponents\LivewirePowerGrid\Helpers\Actions;
+use PowerComponents\LivewirePowerGrid\Themes\{Bootstrap5, Tailwind};
+
+beforeEach(function () {
+    Button::macro('icon', function (string $name) {
+        $this->dynamicProperties['icon'] = $name;
+
+        return $this;
+    });
+});
+
+it('properly handles macroable button method', function (string $theme) {
+    $button = Button::add('edit')
+        ->class('text-center')
+        ->icon('fa-user');
+
+    $actionClass = new Actions(
+        $button,
+        new stdClass(),
+        'id',
+        new $theme(),
+    );
+
+    expect($actionClass->getDynamicProperty('icon'))
+        ->toBe('fa-user');
+})->with('dynamic-action')->group('dynamic-action');
+
+dataset('dynamic-action', [
```

Comment:
Tests cannot be done directly in the themes class, it must be a livewire component like these:

![image](https://user-images.githubusercontent.com/33601626/203339226-b433b921-1434-4cb7-b996-d37fb3239c38.png)

Reply:
@luanfreitasdev in order to transform this test to Livewire component I must create custom template. I think that to test this functionality it is more logical to create unit test rather than creating feature test. What do you think?

Reply:
I think a custom component is better

Reply:
Ok, I will take a look at component with custom template. But it is more logical to use unit tests for this feature because real usage is implementation defined.

Reply:
@martynaskre , can you send a unit test so I can share this feature?